test/unit/test_store.rb

```diff
@@ -102,6 +102,22 @@ def test_create_missing_translations_in_top_level_file
     assert_nil translation.text
   end
 
+  def test_create_missing_keys_uses_replace_locale_in_path_transformation
+    calls = []
+    @store.define_singleton_method(:replace_locale_in_path) do |*args|
+      calls << args
+      "/tmp/special_path/en.yml"
+    end
+
+    @store.add_translation Translation.new(name: "da.app_name", text: "Oversætter", file: "/tmp/da.yml")
+    @store.add_locale("en")
+    @store.create_missing_keys
+
+    assert_equal(1, calls.count)
+    assert_equal(['da', 'en', '/tmp/da.yml'], calls.first)
```

Comment:
I am testing, that the interface defined by `Transformation. replace_locale_in_path` is used and used correctly. Otherwise I can't be sure, that it is actually used and not being replaced with something inline, which would make my encapsulated tests obsolete.